src/widget/wtracktableview.cpp

```diff
@@ -1975,6 +1984,18 @@ void WTrackTableView::slotReloadCoverArt() {
     }
 }
 
+void WTrackTableView::slotSortingChanged(int headerSection, Qt::SortOrder order) {
+    double sortOrder = (order == Qt::AscendingOrder) ? 0.0 : 1.0;
+    if (headerSection != (int)m_pSortColumn->get()) {
```

Comment:
Invoking both slots one after another might sort the whole table twice.

Reply:
The revised version of this handler should update both m_pSortOrder and m_pSortColumn. If at least one of them has been modified it should invoke `afterSortingChanged()`.

Reply:
New member function `applySorting()`  without parameters:
- Read and translate the current values stored in `m_pSortColumn` and `m_pSortOrder`
- Invoke `horizontalHeader()->setSortIndicator()`
- Invoke `doSortByColumn()`

With this strategy the translation of the CO values can be centralized at a single place in the code and all 3 slots could be stripped down to a minimum lines of code.

Reply:
Good catch, I implemented it in bc7fbab01.